### ISSUE TITLE
Unreachable readiness probes with k3s + Cilium CNI and default NetworkPolicies

### ISSUE THREAD

**Issue by clementnuss** [fluxcd/flux2#1450]
### Describe the bug

After installing the Cilium CNI on my K3s Kubernetes cluster, I found all my fluxcd components in a CrashloopBackoff state: they controllers were not able to elect leaders, and the `/healthz` endpoints were unreachable.
After (quite) some digging, I noticed that removing the 3 automatically created NetworkPolicies (when installing flux with the `flux install` command) from the `flux-system` namespace solves the issue.

Is is Cilium that is too strict with the Network Policies ? Or the Network Policies not permissive enough and preventing the health checks between the pods ? In any case I think future users might encounter the same bug.

### To Reproduce

Steps to reproduce the behaviour:

1. Simply install flux on a Kubernetes cluster
2. Deploy Cilium on this Cluster
3. Most probably, the 4 deployments in flux-system namespace will be in a crashloop state

### Expected behavior

The default Network Policies should be sufficient to allow the pods to communicate and synchronize

### Additional context
Single node Kubernetes Cluster on a Raspberry Pi 4 (Ubuntu / ARM64).
- Kubernetes version: 1.21.1+k3s1 (had the same issue on 1.20.7+k3s1)

Below please provide the output of the following commands:

```cli
flux --version > 0.13.4
flux check
► checking prerequisites
✔ kubectl 1.20.4-dirty >=1.18.0-0
✔ Kubernetes 1.21.1+k3s1 >=1.16.0-0
► checking controllers
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.12.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.13.0
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.10.1
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.12.2
✔ all checks passed
```

**Comment by hiddeco** [comment]
This is likely a k3s specific issue, see earlier reported issues: https://github.com/fluxcd/flux2/issues?q=k3s

**Comment by clementnuss** [comment]
indeed, sounds extremely similar to this previous issue: https://github.com/fluxcd/flux2/issues/294
If someone has time to test this issue on a standard K8s+Cilium deployment, might be worth. Not sure K3s is to blame here

**Comment by ohdearaugustin** [comment]
I'm experiencing the same issue. I have no additional network policies in place

k3s version v1.19.10+k3s1 (72e8196c)

Cilium: 1.9.6

flux version 0.14.2

**Comment by ohdearaugustin** [comment]
The comment from @onedr0p in the issue #294 helped.

So if you use bootstrap flux use the flag `--network-policy=false` when installing flux. Furthermore activate masquerading in cilium.

**Comment by rio** [comment]
I actually ran in to this when doing the exact same k3s + cilium thing but with argocd. It's actually sort of the fault of k3s which deploys it's own network policy controller. Next when cilium sees the argocd/flux/any-other network policies it will enforce them as normal but because of some iptables rules that k3s' network policy controller injects cilium will label any health/readiness probes as coming from "world" and thus dropping the packets according to the network policy. This will obviously make all the probes fail and thus this issue.

I've made some changes to the cilium docs (https://github.com/cilium/cilium/pull/16755) viewable at https://docs.cilium.io/en/v1.10/gettingstarted/k3s/. Basically it means adding `--disable-network-policy` to the k3s server and it should just start working again. No need to fully reinstall the cluster, this is basically treated as a server upgrade from the k3s side.
